### PR TITLE
Fixed Mentor Name

### PR DESCRIPTION
**Pull request by ankitpws**
Fixed Mentor Name

### CODE REVIEW

File: src/services/helper/sessions.js [hunk 1]
```diff
@@ -44,6 +45,9 @@ module.exports = class SessionsHelper {
 				bodyData['endDateUtc'] = moment.unix(bodyData.endDate).utc().format(common.UTC_DATE_TIME_FORMAT)
 			}
 
+			const mentorName = await UserProfileHelper.details('', loggedInUserId)
```
Comment (rakeshSgr) [inline src/services/helper/sessions.js:48]: @ankitpws this is not required, and don't store mentor name in DB.

In response only we need to send the result mentor name

File: src/services/helper/sessions.js [hunk 2]
```diff
@@ -426,14 +432,23 @@ module.exports = class SessionsHelper {
 						}
 					})
 				)
+
+				await Promise.all(
+					sessionDetails[0].data.map(async (session) => {
+						const mentorName = await UserProfileHelper.details('', session.userId)
```
Comment (rakeshSgr) [inline src/services/helper/sessions.js:438]: @ankitpws  get all the users data in one API

File: src/services/helper/sessions.js [hunk 3]
```diff
@@ -426,14 +432,23 @@ module.exports = class SessionsHelper {
...
+						session.mentorName = mentorName.data.result.name
+					})
+				)
 			}
+			console.log(sessionDetails[0].data)
```
Comment (rakeshSgr) [inline src/services/helper/sessions.js:443]: @ankitpws remove this

File: src/services/helper/sessions.js [hunk 4]
```diff
@@ -550,6 +568,10 @@ module.exports = class SessionsHelper {
 					responseCode: 'CLIENT_ERROR',
 				})
 			}
+
```
Comment (rakeshSgr) [inline src/services/helper/sessions.js:571]: @ankitpws check with frontent team, they require mentor name when they do enroll and unenroll the session, If require then only add it.

Comment (rakeshSgr) [reply src/services/helper/sessions.js:571]: @ankitpws sending a email we may require.

Comment (ankitpws) [reply src/services/helper/sessions.js:566]: for Email only we require mentor name @rakeshSgr

File: src/services/helper/sessions.js (hunk 2 above)
Comment (ankitpws) [reply src/services/helper/sessions.js:435]: @rakeshSgr there is no API which will give me all the mentor details

Comment (rakeshSgr) [reply src/services/helper/sessions.js:435]: check this getAllAccountsDetail

File: src/services/helper/sessions.js [hunk 5]
```diff
@@ -426,6 +430,19 @@ module.exports = class SessionsHelper {
 						}
 					})
 				)
+				const userIds = sessionDetails[0].data
+					.map((item) => item.userId.toString())
+					.filter((value, index, self) => self.indexOf(value) === index)
+
+				let mentorDetails = await UserProfileHelper.getAllMentorsDetail(userIds)
```
Comment (rakeshSgr) [inline src/services/helper/sessions.js:437]: @ankitpws change to getListOfUserDetails in enhancement

Comment (rakeshSgr) [reply src/services/helper/sessions.js:437]: @ankitpws we don't require all the fields, can we pass only the required fields from here, take it enahncement

File: src/services/helper/sessions.js [hunk 6]
```diff
@@ -426,6 +430,19 @@ module.exports = class SessionsHelper {
...
+				mentorDetails = mentorDetails.result
+
+				for (let i = 0; i < sessionDetails[0].data.length; i++) {
+					let mentorIndex = mentorDetails.findIndex(
```
Comment (rakeshSgr) [inline src/services/helper/sessions.js:441]: @ankitpws improve this logic, if store as object, then you can easily access.
sample : {
   "1214123" :{ ...data }
}

**Pull request merged**